nightfall-optimist/src/services/transaction-sign-send.mjs

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-param-reassign */
+
+import config from 'config';
+import { web3 } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import { estimateGas, estimateGasPrice } from '@polygon-nightfall/common-files/utils/gas.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { Mutex } from 'async-mutex';
+
+const { GAS, GAS_MULTIPLIER, GAS_ESTIMATE_ENDPOINT, GAS_PRICE, GAS_PRICE_MULTIPLIER } = config;
+
+const nonceMutex = new Mutex();
+
+/**
+ * Create and sign a web3 transaction (tx) object
+ *
+ * @async
+ * @function createSignedTransaction
+ * @param {string} ethPrivateKey Eth private key of the sender to sign the tx
+ * @param {string} from Eth address sending the contents of the tx
+ * @param {string} to Eth address receiving the contents of the tx
+ * @param {string} data The unsigned tx generated by encodeABI
+ * @param {number} value Payment for the tx
+ * @returns {Promise<SignedTransaction>}
+ */
+export async function createSignedTransaction(ethPrivateKey, from, to, data, value = 0) {
+  // Check if web3 ws is available
+  const isListening = await web3.eth.net.isListening();
+  if (!isListening) throw new Error('Web3 ws not listening, try again later');
+
+  logger.debug('Create transaction object...');
+
+  let signedTx;
+  await nonceMutex.runExclusive(async () => {
+    // Get nonce
+    const nonce = await web3.eth.getTransactionCount(from);
```

Comment:
The `getTransactionCount` method returns the count of non-pending transactions, while it doesn't consider the Pending ones, I believe this could cause an issue. This might not being caught on our tests because Ganache is configured with mining block time = 1 second.

Reply:
57ca8b3e Changes
#1356 Trying to test this...